app/controllers/external_controller.rb

```diff
@@ -58,21 +55,25 @@ def create_user
     # Set to pending if registration method is approval
     if registration_method == SiteSetting::REGISTRATION_METHODS[:approval]
       user.pending! if new_user
-      return redirect_to '/pending' if user.pending?
+      return redirect_to pending_path if user.pending?
     end
 
     user.generate_session_token!
     session[:session_token] = user.session_token
 
     # TODO: - Ahmad: deal with errors
-    redirect_location = cookies[:location]
-    cookies.delete(:location)
-    return redirect_to redirect_location if redirect_location&.match?('\A\/rooms\/\w{3}-\w{3}-\w{3}-\w{3}\/join\z')
 
-    redirect_to '/'
+    redirect_location = cookies.delete(:location)
+
+    return redirect_to redirect_location, allow_other_host: false if redirect_location
```

Comment:
Shouldn't we still scope this to only redirect to room join?

Reply:
`return redirect_to redirect_location, allow_other_host: false if redirect_location&.match?('\/rooms\/\w{3}-\w{3}-\w{3}-\w{3}\/join\z'')` ?

Reply:
Yeah